### PR TITLE
update to vite and other modern techniques

### PR DESCRIPTION
**Pull request by joellord**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Mern Stack code for the [Mern Tutorial](https://www.mongodb.com/languages/mern-s
 Create the file `mern/server/config.env` with your Atlas URI and the server port:
 ```
 ATLAS_URI=mongodb+srv://<username>:<password>@sandbox.jadwj.mongodb.net/
+PORT=5000
```
Comment (joellord) [inline README.md]: I think we ought to start using port 5050. Port 5000 is causing some issues on Mac OS with another process (Airplay Receiver)

File: mern/client/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>MERN (MongoDB + Express + React + Node</title>
```
Comment (joellord) [inline mern/client/index.html]: Missing closing parenthesis

File: mern/client/src/components/ModifyRecord.jsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  });
+  const params = useParams();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function fetchData() {
+      const id = params.id?.toString() || undefined;
+      if(!id) return;
+      const response = await fetch(
+        `http://localhost:5000/record/${params.id.toString()}`
```
Comment (joellord) [inline mern/client/src/components/ModifyRecord.jsx]: I'd put the base URL in a config file so one doesn't have to change it if the port needs to change.

File: mern/client/src/components/ModifyRecord.jsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+      return { ...prev, ...value };
+    });
+  }
+
+  // This function will handle the submission.
+  async function onSubmit(e) {
+    e.preventDefault();
+    const person = { ...form };
+    try {
+      // if the id is present, we will set the URL to /record/:id, otherwise we will set the URL to /record.
```
Comment (joellord) [inline mern/client/src/components/ModifyRecord.jsx]: See discussion in Google Doc

File: mern/client/src/components/ModifyRecord.jsx
Comment (joellord) [inline mern/client/src/components/ModifyRecord.jsx:1]: Being picky. Should this file have a different name if it does both Edit and Add?

File: mern/client/src/components/ModifyRecord.jsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                  />
+                </div>
+              </div>
+            </div>
+            <div>
+              <fieldset className="mt-4">
+                <legend className="sr-only">Position Options</legend>
+                <div className="space-y-4 sm:flex sm:items-center sm:space-x-10 sm:space-y-0">
+                  <div className="flex items-center">
+                    <input
```
Comment (joellord) [inline mern/client/src/components/ModifyRecord.jsx]: Should we do the three inputs in a positions.map() function?

File: mern/client/src/components/ModifyRecord.jsx [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+                    >
+                      Senior
+                    </label>
+                  </div>
+                </div>
+              </fieldset>
+            </div>
+          </div>
+        </div>
+        <input
```
Comment (joellord) [inline mern/client/src/components/ModifyRecord.jsx]: Should this be a <button> or is this what the cool kids do nowadays?

File: mern/client/src/components/Navbar.jsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { NavLink } from "react-router-dom";
+
+export default function Navbar() {
+  return (
+    <div>
+      <nav className="flex justify-between items-center mb-6">
+        <NavLink to="/">
+          <img alt="MongoDB logo" className="h-10 inline" src="https://d3cy9zhslanhfa.cloudfront.net/media/3800C044-6298-4575-A05D5C6B7623EE37/4B45D0EC-3482-4759-82DA37D8EA07D229/webimage-8A27671A-8A53-45DC-89D7BF8537F15A0D.png"></img>
```
Comment (joellord) [inline mern/client/src/components/Navbar.jsx:8]: Stupid question. Is there a possibility that this link would change in the future?

File: mern/client/src/components/RecordList.jsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  </tr>
+);
+
+export default function RecordList() {
+  const [records, setRecords] = useState([]);
+
+  // This method fetches the records from the database.
+  useEffect(() => {
+    async function getRecords() {
+      const response = await fetch(`http://localhost:5000/record/`);
```
Comment (joellord) [inline mern/client/src/components/RecordList.jsx]: Same comment with the API URL

File: mern/client/src/main.jsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      },
+    ],
+  },
+  {
+    path: "/create",
+    element: <App />,
+    children: [
+      {
+        path: "/create",
+        element: <ModifyRecord />,
```
Comment (joellord) [inline mern/client/src/main.jsx]: We could set the isEdit/isAdd as a parameter here

File: mern/server/routes/record.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
+import express from "express";
+
+// This will help us connect to the database
+import db from "../db/connection.js";
+
+// This help convert the id from string to ObjectId for the _id.
+import { ObjectId } from "mongodb";
+
+// recordRoutes is an instance of the express router.
```
Comment (joellord) [inline mern/server/routes/record.js]: router is an instance of the express router.

File: mern/client/src/components/Navbar.jsx (hunk 7 above)
Comment (codeSTACKr) [reply mern/client/src/components/Navbar.jsx:8]: Honestly, not sure where this link came from. It was originally in the tutorial. I'll swap it for a github repo asset link.

**Pull request merged**